app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -45,6 +47,23 @@ def self.default_capsule
         end
       end
 
+      def puppet_path
+        self[:puppet_path] || update_puppet_path
+      end
+
+      def update_puppet_path
+        return unless has_feature?(PULP_NODE_FEATURE)
```

Comment:
Or, we could maybe deduplicate the two classes in ProxyAPI since they're basically identical and hit /pulp or /pulpnode depending on which feature's enabled.